richtextfx/src/main/java/org/fxmisc/richtext/skin/TextExt.java

```diff
@@ -0,0 +1,95 @@
+package org.fxmisc.richtext.skin;
+
+import com.sun.javafx.css.converters.PaintConverter;
+import javafx.beans.property.ObjectProperty;
+import javafx.css.CssMetaData;
+import javafx.css.Styleable;
+import javafx.css.StyleableObjectProperty;
+import javafx.css.StyleableProperty;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import javafx.scene.text.Text;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TextExt extends Text {
+
+    private ObjectProperty<Paint[]> backgroundColor = null;
```

Comment:
That the way the JavaFX converter works. I was using Color instead of Paint[] before, but the CSS converter injected a Paint array anyway which led to casting errors along the way.

It's the same way in the internal implementation in the Background class. All I can think of is making the property private and expose a Paint property which hides the implementation details.